### PR TITLE
Themes: Move reducers, selectors, and actions to client/state/themes

### LINKED ISSUE #1600: Themes: Move reducers, actions, and selectors to `client/state`
After #840 and #1363, to comply with #1378.

### PR DESCRIPTION
**Pull request by ockham**
Trying to adapt the directory structure to the existing subdirectories of `client/state`. Also, rename `ThemeConstants` module variables to `ActionTypes`, ES6ify the `action-types` module, and move out the only non-action-type constant to where it makes more sense.

No visual changes. To test -- verify that the Theme Showcase still works. As this is a big change, test thoroughly (use end-to-end tests!)

Fixes #1600

### CODE REVIEW

**Comment by ockham** [comment]
Oops, still need to move tests around.

File: client/state/themes/themes/reducer.js [hunk 1]
```diff
@@ -7,35 +7,35 @@ import transform from 'lodash/object/transform';
...
+import ActionTypes from '../action-types';
 
 const initialState = fromJS( {
 	themes: {},
 	currentSiteId: 0
 } );
 
-function add( newThemes, themes ) {
-	return themes.merge( transform( newThemes, ( result, theme ) => {
+function add( newThemes, ts ) {
```
Comment (ehg) [inline client/state/themes/themes/reducer.js]: What's `ts`? :)

Comment (ockham) [inline client/state/themes/themes/reducer.js]: Sry, in a previous iteration `themes`would've shadowed a function name. Renamed back to `themes` in 3f93bebf328a4c51d16cea380d6d89dc55c77561.

**Comment by ehg** [comment]
Tested with multisite e2e tests & logged out, LGTM!

**Comment by ockham** [comment]
Thanks! Squashed, will merge once CirclCI has finished.

**Pull request merged**